### PR TITLE
Propagate Dex version from build args

### PR DESCRIPTION
**Pull request by nabokihms**
#### Overview

Propagate the image version from the image metadata annotations to Dex. It will help us to avoid depending on git analysis by the script and take the date from the CI system.

#### What this PR does / why we need it

Closes https://github.com/dexidp/dex/issues/3180

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -29,7 +29,10 @@ RUN go mod download
 
 COPY . .
 
-RUN make release-binary
+# Propagate Dex version from build args to the build environment
+ARG VERSION
+RUN ([ "$VERSION" != "" ] || export $VERSION) && make release-binary
```
Comment (sagikazarmark) [inline Dockerfile:34]: I don't think you need to export it. It should be available as an environment variable.

Comment (nabokihms) [reply Dockerfile:34]: Tried this without exporting, and it didn't work 🤷 

It works if I do `ENV VERSION=$VERSION`, but returns an empty version if there is no build arg.

Comment (sagikazarmark) [reply Dockerfile:34]: Here is a dummy draft that proofs that it should be working: https://github.com/dexidp/dex/pull/3277

If it doesn't, something else is going on.

Can you please check again?

Comment (nabokihms) [reply Dockerfile:34]: Yep, seems like you are right. I tried the most minimalistic script, and it worked. I will revert the `expose` command instructions.

**Review by sagikazarmark — APPROVED** [review]

**Pull request merged**